pkg/controller/association/ca.go

```diff
@@ -24,40 +22,40 @@ type CASecret struct {
 	CACertProvided bool
 }
 
-// ElasticsearchCACertSecretName returns the name of the secret holding the certificate chain used
-// by the associated resource to establish and validate a secured HTTP connection to Elasticsearch.
-func ElasticsearchCACertSecretName(associated commonv1.Associated, associationName string) string {
+// CACertSecretName returns the name of the secret holding the certificate chain used
+// by the associated resource to establish and validate a secured HTTP connection to the target service.
+func CACertSecretName(associated commonv1.Associated, associationName string) string {
 	return associated.GetName() + "-" + associationName + "-ca"
 }
 
-// ReconcileCASecret keeps in sync a copy of the Elasticsearch CA.
-// It is the responsibility of the controller to set a watch on the ES CA.
-func (r *Reconciler) ReconcileCASecret(associated commonv1.Associated, es types.NamespacedName) (CASecret, error) {
-	publicESHTTPCertificatesNSN := certificates.PublicCertsSecretRef(esv1.ESNamer, es)
+// ReconcileCASecret keeps in sync a copy of the target service CA.
+// It is the responsibility of the association controller to set a watch on this CA.
+func (r *Reconciler) ReconcileCASecret(association commonv1.Association, namer name.Namer, service types.NamespacedName, caSecretServiceName string) (CASecret, error) {
+	servicePublicHTTPCertificatesNSN := certificates.PublicCertsSecretRef(namer, service)
```

Comment:
Not introduced in this PR but I wonder whether we ever considered not copying the secret around if both resources are in the same namespace to reduce the number of secrets we create (and given that this will probably be the common case)

Reply:
👍 Since this PR is pretty large I'd rather work on it in a dedicated PR.

Reply:
Absolutely!